lib/addons/ebs-csi-driver/index.ts

```diff
@@ -1,49 +1,130 @@
 import { PolicyDocument } from "aws-cdk-lib/aws-iam";
 import * as kms from "aws-cdk-lib/aws-kms";
 import { ClusterInfo } from "../../spi";
-import { CoreAddOn } from "../core-addon";
+import { CoreAddOn, CoreAddOnProps } from "../core-addon";
 import { getEbsDriverPolicyDocument } from "./iam-policy";
 import { supportsALL } from "../../utils";
+import { Construct } from "constructs";
+import { KubernetesManifest, KubernetesPatch } from "aws-cdk-lib/aws-eks";
 
 /**
  * Interface for EBS CSI Driver EKS add-on options
  */
-export interface EbsCsiDriverAddOnProps {
-    /**
-     * Version of the driver to deploy
-     */
-    version?: string;
-    /**
-     * List of KMS keys to be used for encryption
-     */
-    kmsKeys?: kms.Key[];
+export interface EbsCsiDriverAddOnProps extends CoreAddOnProps {
+  /**
+   * List of KMS keys to be used for encryption
+   */
+  kmsKeys?: kms.Key[];
+  /**
+   * StorageClass to be used for the addon
+   */
+  storageClass?: string;
 }
 
 /**
  * Default values for the add-on
  */
-const defaultProps = {
-    addOnName: "aws-ebs-csi-driver",
-    version: "v1.23.0-eksbuild.1",
-    saName: "ebs-csi-controller-sa",
+const defaultProps: CoreAddOnProps & EbsCsiDriverAddOnProps = {
+  addOnName: "aws-ebs-csi-driver",
+  version: "v1.23.0-eksbuild.1",
+  saName: "ebs-csi-controller-sa",
+  storageClass: "gp2", // Set the default StorageClass to gp3
 };
 
 /**
  * Implementation of EBS CSI Driver EKS add-on
  */
 @supportsALL
 export class EbsCsiDriverAddOn extends CoreAddOn {
+  readonly ebsProps: EbsCsiDriverAddOnProps;
+
+  constructor(readonly options?: EbsCsiDriverAddOnProps) {
+    super({
+        addOnName: defaultProps.addOnName,
+        version: options?.version ?? defaultProps.version,
+        saName: defaultProps.saName
+
+    });
+
+    this.ebsProps = {
+        ...defaultProps,
+        ...options,
+    };
+}
 
-    constructor(readonly options?: EbsCsiDriverAddOnProps) {
-        super({
-            addOnName: defaultProps.addOnName,
-            version: options?.version ?? defaultProps.version,
-            saName: defaultProps.saName
-        });
-    }
 
-    providePolicyDocument(clusterInfo: ClusterInfo) : PolicyDocument {
-        return getEbsDriverPolicyDocument(clusterInfo.cluster.stack.partition, this.options?.kmsKeys);
+  providePolicyDocument(clusterInfo: ClusterInfo): PolicyDocument {
+    return getEbsDriverPolicyDocument(
+      clusterInfo.cluster.stack.partition,
+      this.options?.kmsKeys
+    );
+  }
+
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    const cluster = clusterInfo.cluster;
+    let updateSc: KubernetesManifest;
+
+    if (this.ebsProps.storageClass) {
+      // patch resource on cluster
+      const patchSC = new KubernetesPatch(

```

Comment:
Please make sure `make lint` succeeds. I see warning like variable is assigned but never used.